test/Fencer/Server/Test.hs

```diff
@@ -87,32 +87,53 @@ test_serverResponseNoRules =
 -- gRPC server
 ----------------------------------------------------------------------------
 
+-- | A type combining a logger, a handle for the logging file, a
+-- thread id and an application state.
+type LogIdSt = (Logger.Logger, Handle, ThreadId, AppState)
+
 -- | Start Fencer on the default port.
-createServer :: IO (Logger.Logger, ThreadId)
+createServer :: IO (Logger.Logger, Handle, ThreadId)
 createServer = do
-  (logger, threadId, _) <- createServerAppState
-  pure (logger, threadId)
+  (logger, logHandle, threadId, _) <- createServerAppState
+  pure (logger, logHandle, threadId)
 
 -- | Start Fencer on the default port.
-createServerAppState :: IO (Logger.Logger, ThreadId, AppState)
+createServerAppState :: IO LogIdSt
 createServerAppState = do
   -- TODO: not the best approach. Ideally we should use e.g.
   -- https://hackage.haskell.org/package/tasty-hunit/docs/Test-Tasty-HUnit.html#v:testCaseSteps
   -- but we can't convince @tinylog@ to use the provided step function.
-  logger <- Logger.create (Logger.Path "/dev/null")
+
+  tmpDir <- Temp.getCanonicalTemporaryDirectory
+  -- This opens a temporary file in the ReadWrite mode
+  (loggerPath, logHandle) <- Temp.openTempFile tmpDir "fencer-server.log"
+  -- The handle has to be closed. Otherwise trying to create a logger
+  -- would fail due to a file lock.
+  hClose logHandle
+  logger <- Logger.create (Logger.Path loggerPath)
   appState <- initAppState
   threadId <- forkIO $ runServer logger appState
-  pure (logger, threadId, appState)
+  pure (logger, logHandle, threadId, appState)
 
 -- | Kill Fencer.
-destroyServer :: (Logger.Logger, ThreadId) -> IO ()
-destroyServer (logger, threadId) = do
+destroyServer :: (Logger.Logger, Handle, ThreadId) -> IO ()
+destroyServer (logger, logHandle, threadId) = do
   Logger.close logger
+  hClose logHandle
   killThread threadId
 
 -- | Kill Fencer.
-destroyServerAppState :: (Logger.Logger, ThreadId, AppState) -> IO ()
-destroyServerAppState (logger, threadId, _) = destroyServer (logger, threadId)
+destroyServerAppState :: LogIdSt -> IO ()
+destroyServerAppState (logger, logHandle, threadId, _) =
+  destroyServer (logger, logHandle, threadId)
```

Comment:
This tuples deconstruction-construction looks noisy. I'd just pass the same `LogIdSt` everywhere regardless of whether all its fields are needed or not.